cmd/tools/test/find_altered_tests.go

```diff
@@ -0,0 +1,188 @@
+	"integration":    {"./common/persistence/tests", "./tools/tests", "./temporaltest", "./internal/temporalite"},
+	"functional":     {"./tests"},
+	"functional_ndc": {"./tests/ndc"},
+	"functional_xdc": {"./tests/xdc"},
+}
+
+func main() {
+	var categories multiFlag
+	var sourceRef string
+	var targetRef string
+
+	flag.Var(&categories, "c", "Category of altered tests to find (can specify multiple)")
+	flag.StringVar(&sourceRef, "s", "", "Source Git reference (commit SHA, branch, etc.)")
+	flag.StringVar(&targetRef, "t", "", "Target Git reference (commit SHA, branch, etc.)")
+	flag.Parse()
+
+	if len(categories) == 0 || sourceRef == "" || targetRef == "" {
+		log.Fatalf("Usage: find_altered_tests -c <category1> -c <category2> ... -s <sourceRef> -t <targetRef>")
+	}
+
+	for _, category := range categories {
+		if _, exists := CategoryDirs[category]; !exists {
+			log.Fatalf("Unknown category: %s", category)
+		}
+	}
+
+	modifiedFiles, err := getModifiedTestFiles(sourceRef, targetRef)
+	if err != nil {
+		log.Fatalf("Error getting modified test files: %v", err)
+	}
+
+	// Initialize a map to hold modified suites per category
+	modifiedSuites := make(map[string][]string)
+
+	for _, category := range categories {
+		dirs := CategoryDirs[category]
+		suites, err := findAlteredTestSuites(modifiedFiles, dirs)
+		if err != nil {
+			log.Fatalf("Error finding altered test suites for category %s: %v", category, err)
+		}
+		modifiedSuites[category] = suites
+	}
+
+	// Output in key=value format
+	for category, suites := range modifiedSuites {
+		joinedSuites := strings.Join(suites, "|")
+		fmt.Printf("modified_%s_test_suites=%s\n", category, joinedSuites)
+	}
+}
+
+// multiFlag allows multiple instances of a flag
+type multiFlag []string
+
+func (m *multiFlag) String() string {
+	return strings.Join(*m, ",")
+}
+
+func (m *multiFlag) Set(value string) error {
+	*m = append(*m, value)
+	return nil
+}
+
+// getModifiedTestFiles runs 'git diff' to find modified '_test.go' files between two references
+func getModifiedTestFiles(sourceRef, targetRef string) ([]string, error) {
+	cmd := exec.Command("git", "diff", "--name-only", sourceRef, targetRef)
+	output, err := cmd.Output()
+	if err != nil {
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			return nil, fmt.Errorf("git diff error: %w", exitErr)
+		}
+		return nil, fmt.Errorf("error running git diff: %w", err)
+	}
+	lines := strings.Split(strings.TrimSpace(string(output)), "\n")
+	var testFiles []string
+	for _, line := range lines {
+		if strings.HasSuffix(line, "_test.go") {
+			testFiles = append(testFiles, line)
+		}
+	}
+	return testFiles, nil
+}
+
+// findAlteredTestSuites filters files by the test directories and extracts test suite names from them
+func findAlteredTestSuites(files []string, testDirs []string) ([]string, error) {
+	var testSuites []string
+	suiteSet := make(map[string]struct{})
+
+	testSuiteRegex := regexp.MustCompile(`func\s+(Test[a-zA-Z0-9_]*Suite)\s*\(`)
```

Comment:
can you add it to the comment? Something like "to be detected _test.go file should contain TestXXXXSuite function.
The rest of .go file LGFM, though I can't review GHA file.